models/repo_activity.go

```diff
@@ -269,3 +309,135 @@ func releasesForActivityStatement(repoID int64, fromTime time.Time) *xorm.Sessio
 		And("release.is_draft = ?", false).
 		And("release.created_unix >= ?", fromTime.Unix())
 }
+
+// FillFromGit returns code statistics for acitivity page
+func (stats *CodeActivityStats) FillFromGit(repo *Repository, fromTime time.Time, allBranches bool) error {
```

Comment:
Shouldn't this be inside https://github.com/go-gitea/git ? Or call to some method in it ?

Reply:
I did not want to have activity stats collection from two separate places to do one thing. Also users stats (top users) that will be implemented in other PR later use database queries

Reply:
Since we will move github.com/go-gitea/git inside of gitea, I'm OK for a temporary case. But we may need some tests for this method I think.

Reply:
done